tests/ui/suggestions/type-ascription-instead-of-let.stderr

```diff
@@ -1,18 +1,8 @@
-error[E0425]: cannot find value `temp` in this scope
-  --> $DIR/type-ascription-instead-of-let.rs:5:9
+error: expected identifier, found `:`
+  --> $DIR/type-ascription-instead-of-let.rs:5:13
    |
 LL |         temp: i32 = fun(5i32);
-   |         ^^^^
-   |         |
-   |         not found in this scope
-   |         help: maybe you meant to write an assignment here: `let temp`
+   |             ^ expected identifier
```

Comment:
Weird that we don't suggest `let ` here :-/

Reply:
The current `help: maybe you meant to write an assignment here: `let temp` is produced at here: https://github.com/rust-lang/rust/blob/1a6ae3d692cfb52b21d0f45ba50b659486e53d6c/compiler/rustc_resolve/src/late/diagnostics.rs#L1873

but the function `type_ascription_suggestion` is removed.

So it's a preexisting issue, now we only support this scenario:

```console
4 |     a = 1;
  |     ^
  |
help: you might have meant to introduce a new binding
  |
4 |     let a = 1;
  |     +++
```

I think we should extend to the scenario of `a: i32 = 1`, I will fix it in another PR.